Wikipedia/Code/InsertMediaSearchResultsCollectionViewController.swift

```diff
@@ -0,0 +1,261 @@
+        let result = searchResults[indexPath.item]
+        cell.configure(imageURL: result.thumbnailURL, imageViewDimension: flowLayout.itemSize.width, caption: result.displayTitle)
+        cell.apply(theme: theme)
+    }
+
+    func setImageInfo(_ imageInfo: MWKImageInfo?, for searchResult: InsertMediaSearchResult, at index: Int) {
+        assert(Thread.isMainThread)
+        searchResult.imageInfo = imageInfo
+    }
+
+    override func scrollViewDidScroll(_ scrollView: UIScrollView) {
+        super.scrollViewDidScroll(scrollView)
+        scrollDelegate?.insertMediaSearchResultsCollectionViewControllerScrollViewDidScroll(self, scrollView: scrollView)
+    }
+
+    // MARK: Themeable
+
+    override func apply(theme: Theme) {
+        super.apply(theme: theme)
+        view.backgroundColor = theme.colors.paperBackground
+        collectionView.backgroundColor = theme.colors.paperBackground
+    }
+
+    // MARK: - Empty State
+
+    var emptyViewType: WMFEmptyViewType = .noSearchResults
+
+    final var isEmpty = true
+    final var showingEmptyViewType: WMFEmptyViewType?
+    final func updateEmptyState() {
+        let sectionCount = numberOfSections(in: collectionView)
+
+        var isCurrentlyEmpty = true
+        for sectionIndex in 0..<sectionCount {
+            if self.collectionView(collectionView, numberOfItemsInSection: sectionIndex) > 0 {
+                isCurrentlyEmpty = false
+                break
+            }
+        }
+
+        guard isCurrentlyEmpty != isEmpty || showingEmptyViewType != emptyViewType else {
+            return
+        }
+
+        isEmpty = isCurrentlyEmpty
+
+        isEmptyDidChange()
+    }
+
+    private var emptyViewFrame: CGRect {
+        let insets = scrollView?.contentInset ?? UIEdgeInsets.zero
+        let frame = view.bounds.inset(by: insets)
+        return frame
+    }
+
+    open func isEmptyDidChange() {
+        if isEmpty {
+            wmf_showEmptyView(of: emptyViewType, action: nil, theme: theme, frame: emptyViewFrame)
+            showingEmptyViewType = emptyViewType
+        } else {
+            wmf_hideEmptyView()
+            showingEmptyViewType = nil
+        }
+    }
+
+    override func scrollViewInsetsDidChange() {
+        super.scrollViewInsetsDidChange()
+        wmf_setEmptyViewFrame(emptyViewFrame)
+        let context = UICollectionViewFlowLayoutInvalidationContext()
+        context.invalidateFlowLayoutAttributes = true
+        flowLayout.invalidateLayout(with: context)
+    }
+
+    override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
+        super.traitCollectionDidChange(previousTraitCollection)
+        let selectedIndexPath = collectionView.indexPathsForSelectedItems?.first
+        collectionView.reloadData()
+        if let selectedIndexPath = selectedIndexPath {
+            collectionView.selectItem(at: selectedIndexPath, animated: false, scrollPosition: .centeredVertically)
```

Comment:
The `selectedIndexPath` might no longer be valid if the number of items changed. Even though the number of items shouldn't change here, it'd be good to add a range check

Reply:
[EDIT] removing the `imageViewDimension` probably eliminates the need for overriding this method and reloading data at all, so check that out first.